src/scripts/dungeons/Dungeon.ts

```diff
@@ -2634,13 +2634,23 @@ dungeonList['Granite Cave'] = new Dungeon('Granite Cave',
         ],
         legendary: [
             {loot: 'MediumRestore'},
-            {loot: 'Star Piece'},
+            {loot: 'Star Piece', ignoreDebuff : true},
```

Comment:
This is intentional to allow players a chance of getting a star piece during the quest step. Figured that this wasn't a huge balance change, either since they're still rare with the debuff off.